### ISSUE TITLE
WebSocket closed; The client-token could be invalidated, please re-login

### ISSUE THREAD

**Issue by PNRxA** [gotify/android#129]
I get this issue daily where this error message comes up. I fix it by going into the app and clicking the refresh button in the hamburger menu. It doesn't require another log in, it just refreshes and connects back to the websocket. I had a look through existing issues but couldn't find anything, is this a known problem?

Device: Pixel 4 XL
Android: 10 on latest security patch
Gotify Android version: 2.0.13 from Play Store
Battery optimisation turned off
Gotify Server: 2.0.17 docker behind caddy

| Notification  | Logs |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/16978632/88761451-812fd180-d1b2-11ea-89d0-3481c7fcc7fa.png" width="100%"> | <img src="https://user-images.githubusercontent.com/16978632/88761457-842ac200-d1b2-11ea-8e6c-e11126b53b98.png" width="100%"> |

**Comment by jmattheis** [comment]
I've some ideas why this is caused, probably the only thing that can be done is to add an automatic reconnect.

**Comment by PNRxA** [comment]
It happens very frequently, sometimes every couple of minutes. I'd love for it to automatically reconnect

**Comment by PNRxA** [comment]
@jmattheis With further testing, while this message is showing notification still come through. Is there any troubleshooting you'd like me to perform?

**Comment by jmattheis** [comment]
Yeah, does gotify/server output any errors? The server logs to stdout.

**Comment by PNRxA** [comment]
Caddy (v2) is set up simply as:

```
gotify.mydomain.run {
        reverse_proxy 192.168.87.84:8070
}
```

(run is the TLD if that matters)

I can then access and use Gotify externally on https://gotify.mydomain.run

My stdout logs don't go far enough at the time of writing this but I'll try to get the output when it happens soon.
You said you might have an idea how this is caused, is it reproducible? If so that might help me to get the logs

**Comment by jmattheis** [comment]
My guess would be that the connection read/write timeouts on server side and the connection will be closed. But this should be logged in gotify/server.

**Comment by PNRxA** [comment]
Ok so I reproduced this by switching networks to/from 4G.

My phone shows the WebSocket closed message but still receives notifications that are pushed to Gotify.

The server logs show the following, you can see the message I posted at the bottom:

> [GIN] 2020/07/31 - 17:27:16 | 200 | 25.5µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:27:46 | 200 | 23.532µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:28:17 | 200 | 38.118µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:28:47 | 200 | 26.914µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:29:18 | 200 | 23.913µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 511.797µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 391.542µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 380.82µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 315.766µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 616.914µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 836.845µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 412.32µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 429.732µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 233.887µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 394.26µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 489.817µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 763.643µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 856.651µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.082462ms | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.237384ms | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 246.424µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 264.351µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 458.528µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 299.529µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 468.036µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 641.735µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 511.797µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 391.542µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 380.82µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 315.766µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 616.914µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 836.845µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 412.32µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 429.732µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 233.887µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 394.26µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 489.817µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 763.643µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 856.651µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.082462ms | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.237384ms | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 246.424µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 264.351µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 458.528µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 299.529µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 468.036µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 641.735µs | my.4G.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.077718ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.404801ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.700395ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.992826ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.070614ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.668951ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 2.047709ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 2.398399ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 577.498µs | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 680.92µs | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 781.324µs | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.275938ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.686372ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 2.458442ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 2.485132ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 3.082596ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 3.391247ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 3.603578ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 3.932744ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 921.833µs | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:29 | 200 | 1.235312ms | my.4G.ip.address | GET "/message?limit=10"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 318.911µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 291.397µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 361.51µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 227.624µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 358.892µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 208.283µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 287.503µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 459.465µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 192.723µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 326.711µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 240.254µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 443.401µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 438.044µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 262.67µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 206.603µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 387.947µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 406.457µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 181.804µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 276.83µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 274.495µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 232.138µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 181.811µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 354.855µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 188.468µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 351.135µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 289.257µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 200.793µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 222.418µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 266.411µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 256.579µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 270.705µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 233.606µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 185.374µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 232.956µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 271.416µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 191.479µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 214.685µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 396.293µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 219.975µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 371.69µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 337.686µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 328.548µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 318.911µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 291.397µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 361.51µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 227.624µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 358.892µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 208.283µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 287.503µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 459.465µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 192.723µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 326.711µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 240.254µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 443.401µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 438.044µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 262.67µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 206.603µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 387.947µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 406.457µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 181.804µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 276.83µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 274.495µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 232.138µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 181.811µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 354.855µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 188.468µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 351.135µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 289.257µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 200.793µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 222.418µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 266.411µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 256.579µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 270.705µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 233.606µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 185.374µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 232.956µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 271.416µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 191.479µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 214.685µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 396.293µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 219.975µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 371.69µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 337.686µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> [GIN] 2020/07/31 - 17:29:38 | 200 | 328.548µs | my.wifi.ip.address | GET "/stream?token=CYfEfUcUAF64fIO"
> WebSocket: ReadError websocket: close 1006 (abnormal closure): unexpected EOF
> WebSocket: ReadError websocket: close 1006 (abnormal closure): unexpected EOF
> [GIN] 2020/07/31 - 17:29:48 | 200 | 29.645µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:29:48 | 200 | 29.645µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:30:18 | 200 | 39.558µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:30:49 | 200 | 26.321µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:31:08 | 200 | 20.594798ms | 192.168.87.1 | POST "/message?token=AkOVpZeaH19pS-D"
> [GIN] 2020/07/31 - 17:31:19 | 200 | 27.088µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:31:50 | 200 | 33.788µs | 127.0.0.1 | GET "/health"
> [GIN] 2020/07/31 - 17:32:20 | 200 | 27.429µs | 127.0.0.1 | GET "/health"

**Comment by jmattheis** [comment]
@PNRxA Could you try out this branch: https://github.com/gotify/android/tree/ws-close to check, if this fixes your problem? In case you've no android build environment ping me back and I'll build the apk for you.

**Comment by PNRxA** [comment]
@jmattheis In my brief testing on this branch, it properly disconnected and reconnected after a minute then sent through the notifications. I'll send through an update in about 2 days of real world use to see how it goes. Appreciate the fast update

**Comment by PNRxA** [comment]
@jmattheis I've had a couple app crashes with this new branch where the notification disappeared, here's the app logs for a crash

![Screenshot_20200804-140321.png](https://user-images.githubusercontent.com/16978632/89251861-803beb80-d65b-11ea-88a6-633a11a32128.png)

**Comment by jmattheis** [comment]
This looks pretty much like the OS has killed gotify.

**Comment by PNRxA** [comment]
@jmattheis if that's the case, this branch has resolved this issue. I've had no further websocket problems and the reconnections have worked perfectly

**Comment by cbwang2016** [comment]
I observed same bug

**Comment by eltiel** [comment]
I'd like to add that I encountered this bug too, but noticed that it only happened after I updated to server 2.0.17. It used to work without issues on 2.0.15

**Comment by PNRxA** [comment]
@jmattheis are we able to merge the branch?

**Comment by jmattheis** [comment]
@PNRxA yeah, sorry totally forgot this issue .__.

**Comment by jmattheis** [comment]
Released in 2.0.14
